Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1304,16 +1308,8 @@ static IntRect computeMarqueeSelectionRect(const WebCore::IntPoint& point1, cons
 void UnifiedPDFPlugin::setCurrentSelection(RetainPtr<PDFSelection>&& selection)
 {
     m_currentSelection = WTFMove(selection);
-    if (!m_selectionTrackingData.isActive)
-        commitCurrentSelection(SelectionCommitReason::SelectionIsNoLongerActive);
-}
-
-void UnifiedPDFPlugin::commitCurrentSelection(SelectionCommitReason reason)
-{
-    if (reason == SelectionCommitReason::ReceivedMouseUp && !std::exchange(m_selectionTrackingData.isActive, false))
-        return;
+    updateLayerHierarchy();
```

Comment:
This is a big hammer to just issue a repaint. Just need to call setNeedsDisplay() somewhere, ideally constraining the repaint to just the old/new selection rect, or at least the touched pages only.

Reply:
That's fair. I can do two optimizations here:

1. Constrain the repaint to the bounds of the selection we're painting
2. Constrain the repaint to the touched pages only

Reply:
Information for both should be readily available, let's see

Reply:
I wasn't able to correctly do the transformation from the page space to the content layer space... deferring that work for a not-so-distant future optimization since I don't want this to block us.

https://bugs.webkit.org/show_bug.cgi?id=268927

Reply:
Let me know if you think this may still be too big a hammer for repaint requests.